bioconda_utils/build.py

```diff
@@ -134,7 +134,7 @@ def build(
                       [os.path.join(recipe, 'meta.yaml')]
                 logger.debug('command: %s', cmd)
                 with utils.Progress():
-                    p = utils.run(cmd, env=os.environ)
+                    utils.logging_run(cmd, logger, env=os.environ)
```

Comment:
Why changing the function name? Why the logger argument? We have a global logger object.

Reply:
> Why changing the function name?
I chose to add another function rather than replace the existing one to keep the changes limited to the places where I inspected the surroundings of the calling code. When running an external binary, the case where you want stdout/stderr directly pushed to the logger are overall the exception.

That said, it would be easy enough to reduce this to one function. 

> Why the logger argument? We have a global logger object.
I gave the `logging_run` function the option to pass a logger object because those lines originate from a binary specified by the caller, and not from the `util` module itself. For example, shell output marked as originating from `pkg_test` or `build` is more helpful than all coming from `util`. 

You are right in that it does not have any visible effect here, though. While the logger objects are per module as usual (not global), bioconda-utils has a custom configured StreamHandler that doesn't print the `name` of the log records. => no effect at this time.